nntrainer/layers/layer_internal.h

```diff
@@ -103,16 +108,15 @@ class Layer : public ml::train::Layer {
    * @param[in] in List of Input Tensors taken by this layer
    * @retval    List of Output Tensors
    */
-  virtual sharedConstTensors forwarding(sharedConstTensors in) = 0;
+  virtual void forwarding(sharedConstTensors in = {}) = 0;
```

Comment:
This is really good. Later, input argument can also be removed.

Reply:
but for the loss calculation, we need to get the label. that why in is still there.